### PR TITLE
Plane: fix VTOL loiters and use vtol_position_controller for VTOL_TRANSITION points

### PR DESCRIPTION
**Pull request by IamPete1**
This fixes https://github.com/ArduPilot/ardupilot/issues/18932. We were using vtol_position_controller for 'VTOL_LOITER' waypoints  without ever calling `poscontrol_init_approach`. This was only caught by the internal error if it was the first VTOL point because in that vase the poscontroller was never run, so we were still in `QPOS_NONE`. The same would happen later in a mission, just in that case we would jump straight into whatever QPOS state it was left in missing the internal error. 

We should be doing a better job of clearing up the QPOS state when were finished with it. 

This also makes VTOL transition way-points work the same way as VTOL land. This is a change in behavior. Instead of doing a crash transition on the spot we now transition for the next waypoint. So we will remain in forward flight longer. 

I hope it does not introduce any equally odd bugs, but I have no idea really. 

This took ages to track down. This part of the code is getting so complicated this is is very hard to be sure if it working as expected. 

I think we should spend some time consolidating the VTOL state to one place rather than having is scattered throughout the code. We should try and reset it correctly when were done with it. 

For example in the current code a VTOL takeoff does not count as VTOL flight. 

Forward flight is OK. VTOL takeoff and land is OK. But trying to fly waypoints in VTOL, or do VTOL guided is just a massive mess. 

Basically I think if I have to sit down for more that 2 hours to try and understand how the code works, that is probably a sign that it is too complicated.

### CODE REVIEW

**Comment by robustini** [comment]
It's a good starting point, thanks @IamPete1!

**Comment by robustini** [comment]
I confirm that the problem is resolved, but others are known about transitions.
I open an issue with logs and related video.

**Comment by IamPete1** [comment]
So many circular references between VTOL mode and VTOL postilion controller.......
